src/script/interpreter.cpp

```diff
@@ -2092,6 +2091,105 @@ bool VerifyTaprootCommitment(uint256 &tapleaf_hash,
     return q == q_expected;
 }
 
+#include <logging.h>
+#include <util/strencodings.h>
+
+static bool VerifyScriptType(std::vector<valtype> stack,
+                             const CScript &script_pubkey, uint32_t flags,
+                             const BaseSignatureChecker &checker,
+                             ScriptExecutionMetrics &metrics_out,
+                             ScriptError *serror) {
+    if (script_pubkey.size() == 1 ||
+        !script_pubkey.IsPushOnly(script_pubkey.begin() + 1)) {
+        return set_error(serror, ScriptError::SCRIPTTYPE_MALFORMED_SCRIPT);
+    }
+    if (script_pubkey[1] == OP_1) { // Taproot script version
```

Comment:
You might want to parse this as a CScriptNum

Reply:
If you do that you can do:

switch(scriptType) {
    case TAPROOT_SCRIPTTYPE:
         VerifyTaproot()...
         break;
    default:
         return set_error(serror, ScriptError::SCRIPTTYPE_INVALID_TYPE);
}